YGOProAnalyticsServer/Controllers/DecklistController.cs

```diff
@@ -19,15 +22,62 @@ public class DecklistController : ControllerBase
         readonly YgoProAnalyticsDatabase _db;
         readonly IDecklistToDecklistDtoConverter _decklistToDtoConverter;
         readonly IDecklistService _decklistService;
+        readonly IAdminConfig _config;
+        readonly IMapper _mapper;
+        readonly IDecklistBrowserQueryParametersDtoValidator _decklistBrowserQueryParamsValidator;
 
         public DecklistController(
             YgoProAnalyticsDatabase db,
             IDecklistToDecklistDtoConverter decklistToDtoConverter,
-            IDecklistService decklistService)
+            IDecklistService decklistService,
+            IAdminConfig config,
+            IMapper mapper,
+            IDecklistBrowserQueryParametersDtoValidator decklistBrowserQueryParamsValidator)
         {
             _db = db ?? throw new ArgumentNullException(nameof(db));
             _decklistToDtoConverter = decklistToDtoConverter ?? throw new ArgumentNullException(nameof(decklistToDtoConverter));
             _decklistService = decklistService ?? throw new ArgumentNullException(nameof(decklistService));
+            _config = config ?? throw new ArgumentNullException(nameof(config));
+            _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+            _decklistBrowserQueryParamsValidator = decklistBrowserQueryParamsValidator 
+                ?? throw new ArgumentNullException(nameof(decklistBrowserQueryParamsValidator));
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> FindAll([FromQuery] DecklistBrowserQueryParametersDTO queryParams)
+        {
+            if (!_decklistBrowserQueryParamsValidator.IsValid(queryParams))
+            {
+                return BadRequest("Invalid params. Please report it to administrator.");
+            }
+
+            DateTime? statisticsFrom = null;
+            DateTime? statisticsTo = null;
+            if (!string.IsNullOrEmpty(queryParams.StatisticsFromDate))
+            {
+                statisticsFrom = DateTime.Parse(queryParams.StatisticsFromDate);
```

Comment:
DateTime Parse may be an issue if the front and backend use different DateTime models